mesa/mesaPROTON_OC.py

```diff
@@ -426,6 +432,147 @@ def setup_siblings(self):
+                        self.families.append(hh_members)
+                    else:
+                        for member in hh_members:
+                            self.population.append(member)
+            if not success:
+                self.complex_hh_sizes.append(size)
+        print("Complex size: " + str(len(self.complex_hh_sizes)) + str("/") + str(len(self.hh_size)))
+        for comp_hh_size in self.complex_hh_sizes:
+            comp_hh_size = int(min(comp_hh_size, len(self.population)))
+            complex_hh_members = self.population[0:comp_hh_size]
+            max_age_index = [x.age() for x in complex_hh_members].index(max([x.age() for x in complex_hh_members]))
+            family_wealth_level = complex_hh_members[max_age_index].wealth_level
+            for member in complex_hh_members:
+                self.population.remove(member)
+                member.makeHouseholdLinks(complex_hh_members)
+                member.wealth_level = family_wealth_level
+            self.families.append(complex_hh_members)
+        print("Singles " + str(len([x for x in self.hh_size if x == 1])))
+        print("Families " + str(len(self.families)))
+
+    def household_sizes(self, size):
+        """
+        loads a .csv with a probability distribution of household size and calculates household based on initial agents
+        :param size: int, the population size, initial agents
+        :return: list, the sizes of household
+        """
+        hh_size_dist = self.read_csv_city("household_size_dist").values.tolist()
+        sizes = []
+        current_sum = 0
+        while current_sum < size:
+            hh_size = extra.pick_from_pair_list(hh_size_dist, self.rng)[0]
+            if current_sum + hh_size <= size:
+                sizes.append(hh_size)
+                current_sum += hh_size
+        sizes.sort(reverse=True)
+        return sizes
+
+    def pick_from_population_pool_by_age_and_gender(self, age_wanted, male_wanted):
+        """
+        Pick an agent with specific age and sex, None otherwise
+        :param age_wanted: int, age wanted
+        :param male_wanted: bool,
+        :return: agent, or None
+        """
+        if male_wanted == True:
+            male_wanted = 1
+        if male_wanted == False:
+            male_wanted = 0
+        if not [x for x in self.population if x.gender == male_wanted and np.floor(x.age()) == age_wanted]:
+            return None
+        picked_person = self.rng.choice(
+            [x for x in self.population if x.gender == male_wanted and np.floor(x.age()) == age_wanted])
+        self.population.remove(picked_person)
+        return picked_person
+
+    def pick_from_population_pool_by_age(self, age_wanted):
+        """
+        Pick an agent with specific age form population, None otherwise
+        :param age_wanted: int, age wanted
+        :return: agent or None
+        """
+        if age_wanted not in [np.floor(x.age()) for x in self.population]:
```

Comment:
At the moment the "Person.age() " method returns a float and the age_wanted parameter is an integer. 
We need the floor unless we change the "Person.age() " method so that it always returns an integer.